service_discovery/README.md

```diff
@@ -0,0 +1,196 @@
+# Service Discovery
+
+- [Introduction](#introduction)
+- [High-level Design](#high-level-design)
+- [Getting Started](#getting-started)
+  - [Prerequisites](#prerequisites)
+  - [Building](#building)
+  - [Running the Tests](#running-the-tests)
+- [Running the Service](#running-the-service)
+  - [Sample Applications](#sample-applications)
+- [Developing with Service Discovery](#developing-with-service-discovery)
+- [Trademarks](#trademarks)
+
+## Introduction
+
+This Service Discovery system for Eclipse Chariott aims to simplify in-vehicle application development by abstracting out where services are running, so that applications that want to leverage the resources and capabilities of another application can discover the location and metadata necessary to communicate. This is a new version of Service Discovery which is decoupled from the intent broker, and it is currently under development. If you would like to use the existing version of Chariott with intent brokering, please refer to the [top-level Readme](./../README.md) for details.
```

Comment:
Here and elsewhere, You have some paragraphs that get broken up into multiple lines, and then you have some paragraphs like here where it is all written on one line. I would recommend splitting them up into multiple lines for readability of the raw markdown file. I picked 100 characters as the limit for my markdown if that helps.